core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/csv.kt

```diff
@@ -345,7 +345,7 @@ public fun DataFrame.Companion.readDelim(
     }
 
     val columnNames = csvParser.headerNames.takeIf { it.isNotEmpty() }
-        ?: (1..records[0].count()).map { index -> "X$index" }
+        ?: (1..(records.firstOrNull()?.count() ?: 0)).map { index -> "X$index" }
```

Comment:
If records.firstOrNull() == null, this will be (1..0)?

Reply:
actually that will be an empty range :)

Reply:
`IntRange.isEmpty()` is actually defined as `first > last` and `IntRange.EMPTY = IntRange(1, 0)`

Reply:
Didn't know that. Ok, cool!